src/app.rs

```diff
@@ -1,112 +1,101 @@
-use std::rc::Rc;
-
 use servo::{
     compositing::{
         windowing::{EmbedderEvent, EmbedderMethods},
         CompositeTarget,
     },
-    embedder_traits::{Cursor, EmbedderMsg, EventLoopWaker},
+    embedder_traits::EventLoopWaker,
+    msg::constellation_msg::BrowsingContextId,
     servo_url::ServoUrl,
     Servo, TopLevelBrowsingContextId,
 };
-use winit::{
-    event::Event,
-    event_loop::{ControlFlow, EventLoopProxy, EventLoopWindowTarget},
-    window::{CursorIcon, Window},
-};
+use winit::{event::Event, event_loop::EventLoopProxy, window::Window as WinitWindow};
 
-use crate::{prefs, resources, webview::WebView};
+use crate::{
+    prefs, resources,
+    window::{GLWindow, Window},
+};
 
-/// Status of webview.
+/// Status of Verso instance.
 #[derive(Clone, Copy, Debug, Default)]
 pub enum Status {
-    /// Nothing happed to this webview yet
+    /// Nothing important to Verso at the moment.
     #[default]
     None,
-    /// Loading of webview has started.
-    LoadStart,
-    /// Loading of webivew has completed.
-    LoadComplete,
+    /// One of webviews is animating.
+    Animating,
     /// Verso has shut down.
     Shutdown,
 }
 
 /// Main entry point of Verso browser.
 pub struct Verso {
-    servo: Option<Servo<WebView>>,
-    webview_id: Option<TopLevelBrowsingContextId>,
-    webview: Rc<WebView>,
+    servo: Option<Servo<GLWindow>>,
+    window: Window,
     events: Vec<EmbedderEvent>,
-    // TODO following fields should move to webvew
     status: Status,
 }
 
 impl Verso {
     /// Create an Verso instance from winit's window and event loop proxy.
-    pub fn new(window: Window, proxy: EventLoopProxy<()>) -> Self {
+    pub fn new(window: WinitWindow, proxy: EventLoopProxy<()>) -> Self {
         resources::init();
         prefs::init();
 
-        let webview = Rc::new(WebView::new(window));
+        let mut window = Window::new(window);
         let callback = Box::new(Embedder(proxy));
         let mut init_servo = Servo::new(
             callback,
-            webview.clone(),
+            window.gl_window(),
             Some(String::from(
                 "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/119.0",
             )),
             CompositeTarget::Fbo,
         );
-
-        let url = ServoUrl::parse("https://servo.org/").unwrap();
-        init_servo
-            .servo
-            .handle_events(vec![EmbedderEvent::NewWebView(url, init_servo.browser_id)]);
+        window.set_webview_id(init_servo.browser_id);
+
+        let demo_path = std::env::current_dir().unwrap().join("demo.html");
+        let demo_url = ServoUrl::from_file_path(demo_path.to_str().unwrap()).unwrap();
+        let demo_id = TopLevelBrowsingContextId::new();
+        let url = ServoUrl::parse("https://wusyong.github.io/").unwrap();
```

Comment:
URL could be some Verso related, like the slides: https://european-browser.github.io/slides/